ui/src/app/views/registration-dashboard/component.tsx

```diff
@@ -48,20 +48,68 @@ const styles = reactStyles({
   }
 });
 
+function redirectToNiH(): void {
+  const url = environment.shibbolethUrl + '/link-nih-account?redirect-url=' +
+          encodeURIComponent(
+            window.location.origin.toString() + '/nih-callback?token={token}');
+  window.location.assign(url);
+}
+
+async function redirectToTraining() {
+  await profileApi().updatePageVisits({page: 'moodle'});
+  window.location.assign(environment.trainingUrl + '/static/data-researcher.html?saml=on');
+}
+
+export const RegistrationTasks = [
+  {
+    title: 'Turn on Google 2-Step Verification',
+    description: 'With 2-Step Verification, you’ll protect your ' +
+      'account with both your password and your phone',
+    buttonText: 'Get Started',
+    completedText: 'Completed',
+    isRefreshable: true,
+    onClick: () => window.open('https://myaccount.google.com/security', '_blank')
+  }, {
+    title: 'Complete Online Training',
+    description: 'Researchers must maintain up-to-date completion of compliance ' +
+      'training courses hosted at the NNLM\'s Moodle installation',
+    buttonText: 'Complete training',
+    completedText: 'Completed',
+    onClick: redirectToTraining
+  }, {
+    title: 'Login to ERA Commons',
+    description: 'Researchers must maintain up-to-date completion of compliance' +
+      ' training courses hosted at the NNLM’s Moodle installation',
+    buttonText: 'Login',
+    completedText: 'Linked',
+    onClick: redirectToNiH
+  }, {
+    title: 'Data Use Agreement',
+    description: 'This data use agreement describes how All of Us ' +
+      'Research Program data can and cannot be used',
+    buttonText: 'View & Sign',
+    completedText: 'Signed',
+    onClick: () => navigate(['data-use-agreement'])
+  }
+];
+
 export interface RegistrationDashboardProps {
   betaAccessGranted: boolean;
-  dataUseAgreementCompleted: boolean;
   eraCommonsLinked: boolean;
   eraCommonsError: string;
   trainingCompleted: boolean;
-  twoFactorAuthCompleted: boolean;
   firstVisitTraining: boolean;
+  twoFactorAuthCompleted: boolean;
+  dataUseAgreementCompleted: boolean;
 }
 
-export class RegistrationDashboard extends
-    React.Component<RegistrationDashboardProps,
-    {trainingWarningOpen: boolean, showRefreshButton: boolean,
-      taskCompletionMap: Map<number, boolean>}> {
+interface State {
+  showRefreshButton: boolean;
```

Comment:
don't think this needs to be defined above.  adds more boilerplate imo

Reply:
Discussion involving `Props` applies.